Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,768 @@
+      known. This flag currently has no effect.
+
+   .. attribute:: checksum_flag
+
+      A 4-byte checksum using XXHash64 of the uncompressed content is written
+      at the end of each frame. Zstandard's decompression code verifies the
+      checksum. If there is a mismatch a :class:`ZstdError` exception is
+      raised.
+   .. attribute:: dict_id_flag
+
+      When compressing with a :class:`ZstdDict`, the dictionary's ID is written
+      into the frame header.
+
+   .. attribute:: nb_workers
+
+      Select how many threads will be spawned to compress in parallel. When
+      :attr:`~.nb_workers` >= 1, enables multi-threaded compression, 1
+      means "1-thread multi-threaded mode". More workers improve speed, but
+      also increases memory usage and slightly reduce compression ratio.
+
+   .. attribute:: job_size
+
+      Size of a compression job, in bytes. This value is enforced only when
+      :attr:`~CompressionParameter.nb_workers` >= 1. Each compression job is
+      completed in parallel, so this value can indirectly impact the number of
+      active threads.
+
+   .. attribute:: overlap_log
+
+      Sets how much data is reloaded from previous jobs (threads) for new jobs
+      to be used by the look behind window during compression. This values is
+      only used when :attr:`~CompressionParameter.nb_workers` >= 1. Acceptable
+      values vary from 0 to 9.
+
+         * 0 means dynamically set the overlap amount
+         * 1 means no overlap
+         * 9 means use a full window size from the previous job
+
+      Each increment halves/doubles the overlap size. "8" means an overlap of
+      ``window_size/2``, "7" means an overlap of ``window_size/4``, etc.
+
+.. class:: DecompressionParameter()
+
+   An :class:`~enum.IntEnum` containing the advanced decompression parameter
+   names that can be used when decompressing data.
+
+   The :meth:`~.bounds` method can be used on any attribute to get the valid
+   values for that parameter.
+
+   .. method:: bounds()
+
+      Return the tuple of int bounds, ``(lower, upper)``, of a decompression
+      parameter. This method should be called on the attribute you wish to
+      retrieve the bounds of. For example, to get the valid values for
+      :attr:`~.window_log_max`, one may check the result of
+      ``CompressionParameter.window_log_max.bounds()``.
+
+      Both the lower and upper bounds are inclusive.
+
+   .. attribute:: window_log_max
+
+      The power of two maximum size of the window used during decompression.
+      This can be useful to limit the amount of memory used when decompressing
+      data.
+
+.. class:: Strategy()
+
+   An :class:`~enum.IntEnum` containing strategies for compression.
+   Higher-numbered strategies correspond to more complex and slower
+   compression.
+
+   .. note::
+
+      The values of attributes of :class:`Strategy` are not necessarily stable
+      between zstd versions. Only the ordering may be relied upon.
```

Comment:
I think we should rephrase, there shouldn't be an expectation that the Python user is able to control their zstd version (e.g. Windows, or a linux redistribuition)

Reply:
Yeah this should be rephrased. I don't mean to imply that the user should change their zstd version, merely that across different versions (different Python versions/OS/distro/etc.) the values could theoretically not be stable.

Reply:
I tried modifying this a little. If it's still unclear I can try again.